pkg/metrics/metrics.go

```diff
@@ -0,0 +1,11 @@
+package metrics
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// Interface represents metrics interface
+type Interface interface {
+	Close()
+	EmitFloat(string, float64, ...string) error
```

Comment:
instead of ...string, why not map[string]string?

Reply:
It looks and feelsmuch nicer and cleaner when constructing metrics calls:
```
defer m.EmitFloat("frontend.duration",
				float64(time.Now().Sub(t)/time.Millisecond),
				"verb", r.Method,
				"api-version", vars["api-version"],
				"code", strconv.Itoa(w.(*logResponseWriter).statusCode),
				"route", routeName)
```
And works just the same. 
```
defer m.EmitFloat("frontend.duration",float64(time.Now().Sub(t)/time.Millisecond))
```